### ISSUE TITLE
Can't create pets, exception thrown.

### ISSUE THREAD

**Issue by CharlieIsHere** [TechzoneMC/SonarPet#9]
````
[11:48:08] [Server thread/INFO]: CharlieIsHere issued server command: /pet sheep
[11:48:08] [Server thread/ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing 'pet sheep' in com.dsh105.echopet.commands.util.DynamicPluginCommand(pet)
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155) ~[Spigogt.jar:git-Spigogt-ead9aa1]
        at org.bukkit.craftbukkit.v1_8_R3.CraftServer.dispatchCommand(CraftServer.java:666) ~[Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.PlayerConnection.handleCommand(PlayerConnection.java:1194) [Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:1006) [Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.PacketPlayInChat.a(PacketPlayInChat.java:45) [Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.PacketPlayInChat.a(PacketPlayInChat.java:1) [Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.NetworkManager.a(NetworkManager.java:277) [Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.ServerConnection.c(ServerConnection.java:154) [Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:832) [Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:381) [Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:733) [Spigogt.jar:git-Spigogt-ead9aa1]
        at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:636) [Spigogt.jar:git-Spigogt-ead9aa1]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_74]
Caused by: com.dsh105.echopet.compat.api.registration.PetRegistrationException: java.lang.IllegalStateException: Failed to create pet object for CharlieIsHere
        at com.dsh105.echopet.compat.api.registration.PetRegistry.performRegistration(PetRegistry.java:121) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry.spawn(PetRegistry.java:102) ~[?:?]
        at com.dsh105.echopet.compat.api.entity.PetType.getNewPetInstance(PetType.java:147) ~[?:?]
        at com.dsh105.echopet.api.PetManager.createPet(PetManager.java:121) ~[?:?]
        at com.dsh105.echopet.commands.PetCommand.onCommand(PetCommand.java:411) ~[?:?]
        at com.dsh105.echopet.commands.util.DynamicPluginCommand.execute(DynamicPluginCommand.java:46) ~[?:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:146) ~[Spigogt.jar:git-Spigogt-ead9aa1]
        ... 12 more
Caused by: java.lang.IllegalStateException: Failed to create pet object for CharlieIsHere
        at com.dsh105.echopet.compat.api.registration.PetRegistrationEntry.createFor(PetRegistrationEntry.java:85) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry$1.call(PetRegistry.java:105) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry$1.call(PetRegistry.java:102) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry.performRegistration(PetRegistry.java:119) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry.spawn(PetRegistry.java:102) ~[?:?]
        at com.dsh105.echopet.compat.api.entity.PetType.getNewPetInstance(PetType.java:147) ~[?:?]
        at com.dsh105.echopet.api.PetManager.createPet(PetManager.java:121) ~[?:?]
        at com.dsh105.echopet.commands.PetCommand.onCommand(PetCommand.java:411) ~[?:?]
        at com.dsh105.echopet.commands.util.DynamicPluginCommand.execute(DynamicPluginCommand.java:46) ~[?:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:146) ~[Spigogt.jar:git-Spigogt-ead9aa1]
        ... 12 more
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_74]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_74]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_74]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_74]
        at com.dsh105.echopet.compat.api.registration.PetRegistrationEntry.createFor(PetRegistrationEntry.java:77) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry$1.call(PetRegistry.java:105) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry$1.call(PetRegistry.java:102) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry.performRegistration(PetRegistry.java:119) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry.spawn(PetRegistry.java:102) ~[?:?]
        at com.dsh105.echopet.compat.api.entity.PetType.getNewPetInstance(PetType.java:147) ~[?:?]
        at com.dsh105.echopet.api.PetManager.createPet(PetManager.java:121) ~[?:?]
        at com.dsh105.echopet.commands.PetCommand.onCommand(PetCommand.java:411) ~[?:?]
        at com.dsh105.echopet.commands.util.DynamicPluginCommand.execute(DynamicPluginCommand.java:46) ~[?:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:146) ~[Spigogt.jar:git-Spigogt-ead9aa1]
        ... 12 more
Caused by: java.lang.NoClassDefFoundError: Could not initialize class net.techcable.sonarpet.particles.Particle
        at com.dsh105.echopet.compat.nms.v1_8_R3.NMSImpl.spawn(NMSImpl.java:59) ~[?:?]
        at com.dsh105.echopet.compat.nms.v1_8_R3.NMSImpl.spawn(NMSImpl.java:35) ~[?:?]
        at com.dsh105.echopet.api.pet.Pet.<init>(Pet.java:72) ~[?:?]
        at com.dsh105.echopet.api.pet.type.SheepPet.<init>(SheepPet.java:36) ~[?:?]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_74]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_74]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_74]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_74]
        at com.dsh105.echopet.compat.api.registration.PetRegistrationEntry.createFor(PetRegistrationEntry.java:77) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry$1.call(PetRegistry.java:105) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry$1.call(PetRegistry.java:102) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry.performRegistration(PetRegistry.java:119) ~[?:?]
        at com.dsh105.echopet.compat.api.registration.PetRegistry.spawn(PetRegistry.java:102) ~[?:?]
        at com.dsh105.echopet.compat.api.entity.PetType.getNewPetInstance(PetType.java:147) ~[?:?]
        at com.dsh105.echopet.api.PetManager.createPet(PetManager.java:121) ~[?:?]
        at com.dsh105.echopet.commands.PetCommand.onCommand(PetCommand.java:411) ~[?:?]
        at com.dsh105.echopet.commands.util.DynamicPluginCommand.execute(DynamicPluginCommand.java:46) ~[?:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:146) ~[Spigogt.jar:git-Spigogt-ead9aa1]
        ... 12 more
[11:48:09] [Server thread/WARN]: Entity threw exception at world:-228.3303767054429,72.0,-189.06371130262505
[11:48:09] [Server thread/WARN]: java.lang.NullPointerException
[11:48:09] [Server thread/WARN]:        at com.dsh105.echopet.api.pet.Pet.getCraftPet(Pet.java:94)
[11:48:09] [Server thread/WARN]:        at com.dsh105.echopet.api.pet.Pet.getLocation(Pet.java:99)
[11:48:09] [Server thread/WARN]:        at com.dsh105.echopet.compat.nms.v1_8_R3.entity.EntityPet.getLocation(EntityPet.java:133)
[11:48:09] [Server thread/WARN]:        at com.dsh105.echopet.compat.nms.v1_8_R3.entity.ai.PetGoalFollowOwner.tick(PetGoalFollowOwner.java:128)
[11:48:09] [Server thread/WARN]:        at com.dsh105.echopet.compat.api.ai.PetGoalSelector.updateGoals(PetGoalSelector.java:175)
[11:48:09] [Server thread/WARN]:        at com.dsh105.echopet.compat.nms.v1_8_R3.entity.EntityPet.t_(EntityPet.java:459)
[11:48:09] [Server thread/WARN]:        at net.minecraft.server.v1_8_R3.World.entityJoinedWorld(World.java:1654)
[11:48:09] [Server thread/WARN]:        at net.minecraft.server.v1_8_R3.World.g(World.java:1621)
[11:48:09] [Server thread/WARN]:        at net.minecraft.server.v1_8_R3.World.tickEntities(World.java:1453)
[11:48:09] [Server thread/WARN]:        at net.minecraft.server.v1_8_R3.WorldServer.tickEntities(WorldServer.java:600)
[11:48:09] [Server thread/WARN]:        at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:873)
[11:48:09] [Server thread/WARN]:        at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:381)
[11:48:09] [Server thread/WARN]:        at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:733)
[11:48:09] [Server thread/WARN]:        at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:636)
[11:48:09] [Server thread/WARN]:        at java.lang.Thread.run(Thread.java:745)
````

**Comment by Techcable** [comment]
Hmm, I'd like to see what exception caused the NoClassDefFoundError.
Do you have any other NoClassDefFoundError in your logs?

**Comment by Techcable** [comment]
@L4BORG found the original error.
http://pastebin.com/gJ2sT4SK

**Comment by Techcable** [comment]
Fixed by 8cfa1d27b55a8aded40ee81ef701a0e309d37f9d though I don't know if there are other errors on 1.8.
